protocols/protocol/cloudstate/crdt.proto

```diff
@@ -106,79 +144,144 @@ message CrdtStreamOut {
     }
 }
 
+//*CrdtState* TODO
+//
 // The CRDT state. This represents the full state of a CRDT. When received, a user function should replace the current
 // state with this, not apply it as a delta. This includes both for the top level CRDT, and embedded CRDTs, such as
 // the values of an ORMap.
+//
+//---
+//
+//`Specification`:
+//
+//  - A `CrdtState` is considered to be valid iff:
+//    - `state` MUST be one of the following:
+//      - `gcounter` TODO
+//      - `pncounter` TODO
+//      - `gset` TODO
+//      - `orset` TODO
+//      - `lwwregister` TODO
+//      - `flag` TODO
+//      - `ormap` TODO
+//      - `vote` TODO
 message CrdtState {
+
     oneof state {
-        // A Grow-only Counter
+        // A Grow-only Counter.
         GCounterState gcounter = 1;
 
-        // A Positve-Negative Counter
+        // A Positve-Negative Counter.
         PNCounterState pncounter = 2;
 
-        // A Grow-only Set
+        // A Grow-only Set.
         GSetState gset = 3;
 
-        // An Observed-Removed Set
+        // An Observed-Removed Set.
         ORSetState orset = 4;
 
-        // A Last-Write-Wins Register
+        // A Last-Write-Wins Register.
         LWWRegisterState lwwregister = 5;
 
-        // A Flag
+        // A Flag.
         FlagState flag = 6;
 
-        // An Observed-Removed Map
+        // An Observed-Removed Map.
         ORMapState ormap = 7;
 
-        // A vote
+        // A vote.
         VoteState vote = 8;
     }
 }
 
+//*GCounterState* TODO
+//
 // A Grow-only counter
 //
 // A G-Counter can only be incremented, it can't be decremented.
+//
+//---
+//
+//`Specification`:
+//
+//  - A `GCounterState` is considered to be valid iff:
+//    - `value` TODO
 message GCounterState {
 
     // The current value of the counter.
     uint64 value = 1;
 }
 
+//*PNCounterState* TODO
+//
 // A Positve-Negative Counter
 //
 // A PN-Counter can be both incremented and decremented.
+//
+//---
+//
+//`Specification`:
+//
+//  - A `PNCounterState` is considered to be valid iff:
+//    - `value` TODO
 message PNCounterState {
 
     // The current value of the counter.
     int64 value = 1;
```

Comment:
signed integers, if the value is to be expected negative, can be more efficient encoded regarding its size. A small negative int64 is encoded [always 10 bytes long](https://developers.google.com/protocol-buffers/docs/encoding#signed-integers). We could use an `sint64` instead of an `int64` here I think, the same as we do for the delta. And we could be more explicit about the intent.

Reply:
@marcellanz :)

Reply:
@marcellanz sint64 seems like a good idea

Reply:
context from another discussion: https://github.com/cloudstateio/cloudstate/pull/353#discussion_r441907563 where its not advised to use unsigned in (public) API's.